packages/hix/lib/Hix/New.hs

```diff
@@ -167,10 +187,23 @@ newProjectFiles conf = do
     }
     ]
   where
-    pathError = noteEnv "Can't convert project name to file path"
     modName = toText modNameS
     modNameS = pascal (toString conf.name.unProjectName)
 
+initProject :: InitProjectConfig -> M ()
+initProject conf = traverse_ createFile =<< newProjectFiles conf
+
 newProject :: NewProjectConfig -> M ()
-newProject conf =
-  traverse_ createFile =<< newProjectFiles conf
+newProject conf = do
+  directory <- pathError (parseSomeDir (toString conf.directory.unProjectDirectory))
+  let name = prjSomeBase (Text.dropWhileEnd (== '/') . Text.pack . fromRelDir . dirname) directory
+  AppResources { cwd } <- M ask
+  let newCwd = case directory of
+        Abs p -> p
+        Rel p -> cwd </> p
```

Comment:
There's an option parser named [`absFileOrCwdOption`](https://github.com/tek/hix/blob/main/packages/hix/lib/Hix/Optparse.hs#L58) that you can duplicate to get a `Dir` version, which implements exactly this logic!

Reply:
damn this is cool